core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -167,5 +169,17 @@ private static String pathContext(String tableLocation) {
 
       return resolvedContext;
     }
+
+    private static String computeHash(String fileName) {
+      Preconditions.checkState(fileName != null, "fileName cannot be null");
```

Comment:
This would be an argument problem, not a state problem. Also, I don't think the right place for this check is in a private static util method. User-facing methods may want to do this (but don't, so probably not) but it isn't a good idea to check it here.